library/constant_time.c

```diff
@@ -63,8 +63,10 @@
  * only used here.
  */
 #if defined(MBEDTLS_EFFICIENT_UNALIGNED_ACCESS) && defined(MBEDTLS_HAVE_ASM)
-#if ((defined(__arm__) || defined(__thumb__) || defined(__thumb2__)) && (UINTPTR_MAX == 0xfffffffful)) || \
-    (defined(__aarch64__) && ((UINTPTR_MAX == 0xffffffffull) || (UINTPTR_MAX == 0xffffffffffffffffull)))
+#if ((defined(__arm__) || defined(__thumb__) || defined(__thumb2__)) && \
```

Comment:
This is a little hard to read even with correct intendation, and the misleading indentiation makes it worse. Can you find a way to pacify uncrustify? I tried exchanging the pointer size checks and the architecture:
```
#if (UINTPTR_MAX == 0xfffffffful &&                                     \
     (defined(__arm__) || defined(__thumb__) || defined(__thumb2__))) || \
    ((UINTPTR_MAX == 0xffffffffull || UINTPTR_MAX == 0xffffffffffffffffull) && \
     defined(__aarch64__))
```
but uncrustify still wants to indent the last `defined(__aarch64__)` line by one column less. It seems to lose track of the parenthesis nesting level in multiline preprocessor directives.

Our best bet here might be `/* INDENT-OFF (uncrustify bug: TODO) */`. I observed the same behavior in the latest upstream commit so I'm going to file a bug with uncrustify.

Reply:
Maybe we could define something like `PTR_WIDTH` in common.h which would shorten this enough to make uncrustify lay it out nicely?

Reply:
Also, I think `__thumb__` is always defined if `__thumb2__` is defined, so we could probably remove the `__thumb2__` part. But it feels safer / more readable keeping that check explicit.

Reply:
> Is there an uncrustify setting we need to tweak?

Not that I could find. (But of course with hundreds of settings I might have missed one.)

> I think __thumb__ is always defined if __thumb2__ is defined

Yes. `__thumb2__` is a refinement of `__thumb__`.

Reply:
Reported to the uncrustify project: https://github.com/uncrustify/uncrustify/issues/4043

Reply:
This doesn't seem to be related to boolean expressions in parens. Changing the settings related to that doesn't seem to affect it.

We can disable the formatting of lines ending in ` \` by setting `disable_processing_nl_cont` in the config. We restyle a couple of files subtly but not in any major way and it stops restyling this part.

I can make a PR for this if desired?

Reply:
I withdraw my request for a workaround here since [the problem is already prevalent in our code base](https://github.com/Mbed-TLS/mbedtls/issues/7820). We'll just have to live with it.

Reply:
Marking this conversation as resolved